core/types/block_test.go

```diff
@@ -518,7 +518,7 @@ func TestWithdrawalsEncoding(t *testing.T) {
 	var decoded Block
 	require.NoError(t, rlp.DecodeBytes(encoded, &decoded))
 
-	assert.Equal(t, block, &decoded)
```

Comment:
The assert here was comparing strict equality even in the cached hash pointers, but this is not necessary, as long as the structures result in the same hash we can be sure they are equal.